codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/generators/PaginatorGenerator.kt

```diff
@@ -276,12 +275,7 @@ class PaginatorGenerator private constructor(
     }
 
     private fun nextTokenEmpty(token: String): String {
-        val tokenType = model.expectShape(paginationInfo.outputTokenMemberPath.last().target)
-        if (tokenType is StringShape) {
-            return "$token.as_deref().unwrap_or_default().is_empty()"
-        } else {
-            return "$token.is_none()"
-        }
+        return "$token.map(|token|token.is_empty()).unwrap_or(true)"
```

Comment:
I think we should prefer function pointers in place of closures where possible:
```suggestion
        return "$token.map(String::is_empty).unwrap_or(true)"
```
It might not be possible depending on how type inference interprets this though

Reply:
this can be either a string or a hashmap